lib/bolt/plugin/puppet_connect_data.rb

```diff
@@ -3,12 +3,34 @@
 module Bolt
   class Plugin
     class PuppetConnectData
+      INPUT_DATA_VAR = 'PUPPET_CONNECT_INPUT_DATA'
+
       def initialize(context:, **_opts)
-        puppet_connect_data_yaml_path = File.join(context.boltdir, 'puppet_connect_data.yaml')
+        if ENV.key?(INPUT_DATA_VAR)
```

Comment:
I think we should comment here on what what this var changes about the plugin and why. The way the code interacts with the variables set makes sense to me, but leaving the extra breadcrumb of what/why here should be helpful in the future.

Reply:
Done.